src/common-utils/main.sh

```diff
@@ -104,18 +104,22 @@ install_debian_packages() {
 
     # Needed for adding manpages-posix and manpages-posix-dev which are non-free packages in Debian
     if [ "${ADD_NON_FREE_PACKAGES}" = "true" ]; then
-        # Bring in variables from /etc/os-release like VERSION_CODENAME
-        sed -i -E "s/deb http:\/\/(deb|httpredir)\.debian\.org\/debian ${VERSION_CODENAME} main/deb http:\/\/\1\.debian\.org\/debian ${VERSION_CODENAME} main contrib non-free/" /etc/apt/sources.list
-        sed -i -E "s/deb-src http:\/\/(deb|httredir)\.debian\.org\/debian ${VERSION_CODENAME} main/deb http:\/\/\1\.debian\.org\/debian ${VERSION_CODENAME} main contrib non-free/" /etc/apt/sources.list
-        sed -i -E "s/deb http:\/\/(deb|httpredir)\.debian\.org\/debian ${VERSION_CODENAME}-updates main/deb http:\/\/\1\.debian\.org\/debian ${VERSION_CODENAME}-updates main contrib non-free/" /etc/apt/sources.list
-        sed -i -E "s/deb-src http:\/\/(deb|httpredir)\.debian\.org\/debian ${VERSION_CODENAME}-updates main/deb http:\/\/\1\.debian\.org\/debian ${VERSION_CODENAME}-updates main contrib non-free/" /etc/apt/sources.list
-        sed -i "s/deb http:\/\/security\.debian\.org\/debian-security ${VERSION_CODENAME}\/updates main/deb http:\/\/security\.debian\.org\/debian-security ${VERSION_CODENAME}\/updates main contrib non-free/" /etc/apt/sources.list
-        sed -i "s/deb-src http:\/\/security\.debian\.org\/debian-security ${VERSION_CODENAME}\/updates main/deb http:\/\/security\.debian\.org\/debian-security ${VERSION_CODENAME}\/updates main contrib non-free/" /etc/apt/sources.list
-        sed -i "s/deb http:\/\/deb\.debian\.org\/debian ${VERSION_CODENAME}-backports main/deb http:\/\/deb\.debian\.org\/debian ${VERSION_CODENAME}-backports main contrib non-free/" /etc/apt/sources.list
-        sed -i "s/deb-src http:\/\/deb\.debian\.org\/debian ${VERSION_CODENAME}-backports main/deb http:\/\/deb\.debian\.org\/debian ${VERSION_CODENAME}-backports main contrib non-free/" /etc/apt/sources.list
-        # Handle bullseye location for security https://www.debian.org/releases/bullseye/amd64/release-notes/ch-information.en.html
-        sed -i "s/deb http:\/\/security\.debian\.org\/debian-security ${VERSION_CODENAME}-security main/deb http:\/\/security\.debian\.org\/debian-security ${VERSION_CODENAME}-security main contrib non-free/" /etc/apt/sources.list
-        sed -i "s/deb-src http:\/\/security\.debian\.org\/debian-security ${VERSION_CODENAME}-security main/deb http:\/\/security\.debian\.org\/debian-security ${VERSION_CODENAME}-security main contrib non-free/" /etc/apt/sources.list
+        if [ ! -e "/etc/apt/sources.list" ]; then 
```

Comment:
Can you show me the contents of `/etc/apt/sources.list` file before we do string replacement?

Reply:
Also, output of `ls /etc/apt`

Reply:
Tested for `bullseye` with disabled `nonFreePackages` :
```
root ➜ / $ cat etc/apt/sources.list
# deb http://snapshot.debian.org/archive/debian/20240130T194419Z bullseye main
deb http://deb.debian.org/debian bullseye main
# deb http://snapshot.debian.org/archive/debian-security/20240130T194419Z bullseye-security main
deb http://deb.debian.org/debian-security bullseye-security main
# deb http://snapshot.debian.org/archive/debian/20240130T194419Z bullseye-updates main
deb http://deb.debian.org/debian bullseye-updates main
```

Tested for `bullseye` again after enabling `nonFreePackages` :
```
# deb http://snapshot.debian.org/archive/debian/20240130T194419Z bullseye main
deb http://deb.debian.org/debian bullseye main contrib non-free
# deb http://snapshot.debian.org/archive/debian-security/20240130T194419Z bullseye-security main
deb http://deb.debian.org/debian-security bullseye-security main
# deb http://snapshot.debian.org/archive/debian/20240130T194419Z bullseye-updates main
deb http://deb.debian.org/debian bullseye-updates main contrib non-free
```

Reply:
output of ls -la /etc/apt/sources.list.d :

root ➜ / $ ls -la /etc/apt/sources.list.d
total 16
drwxr-xr-x 1 root root 4096 Feb  7 00:29 .
drwxr-xr-x 1 root root 4096 Jan 30 19:44 ..
-rw-r--r-- 1 root root  477 Feb  7 00:29 debian.sources

Reply:
Contents of /etc/apt/sources.list.d/debian.sources before we do string replacement :
```
Types: deb
# http://snapshot.debian.org/archive/debian/20240130T194419Z
URIs: http://deb.debian.org/debian
Suites: bookworm bookworm-updates
Components: main
Signed-By: /usr/share/keyrings/debian-archive-keyring.gpg

Types: deb
# http://snapshot.debian.org/archive/debian-security/20240130T194419Z
URIs: http://deb.debian.org/debian-security
Suites: bookworm-security
Components: main
Signed-By: /usr/share/keyrings/debian-archive-keyring.gpg
```

After replacing strings globally :
```
Types: deb
# http://snapshot.debian.org/archive/debian/20240130T194419Z
URIs: http://deb.debian.org/debian
Suites: bookworm bookworm-updates
components: main contrib non-free
Signed-By: /usr/share/keyrings/debian-archive-keyring.gpg
 
Types: deb
# http://snapshot.debian.org/archive/debian-security/20240130T194419Z
URIs: http://deb.debian.org/debian-security
Suites: bookworm-security
components: main contrib non-free
Signed-By: /usr/share/keyrings/debian-archive-keyring.gpg
```